### DISCUSSION TITLE
Is there a way to use actions centrally in an org?

### DISCUSSION THREAD

**Discussion by marina-chibizova** [community/community#46209] [Actions]
### Select Topic Area

Question

### Body

We have a lot of different repositories and a lot of actions for them. It takes a lot of manual work to add actions /workflows to each repo and set variables and secrets there. Is there a way to avoid doing anything in individual repos and instead manage all actions and variables in one place (on the org level or in a dedicated repo?). Or the only solution is to build an App with all the needed features we use in actions?

**Comment by MostafaWaleed0** [comment]
Yes, you can manage actions and secrets at the organization level in GitHub. This will allow you to define actions and secrets that can be shared across multiple repositories in your organization.

To manage actions at the organization level, you can create a repository dedicated to actions and include the necessary actions and workflows in that repository. Then, you can use the "GitHub Actions for a repository" event trigger in the individual repositories to run the actions in the dedicated repository.

To manage secrets at the organization level, you can use the GitHub Secrets API to store secrets at the organization level and reference them in your workflows.

Alternatively, you can build a GitHub App to manage actions and secrets. The app can define actions and workflows and store secrets that can be shared across multiple repositories in your organization. This approach can provide more customization options and integration with other services, but it requires a more in-depth development effort.

The choice between these options will depend on the specific requirements of your organization and the amount of effort you're willing to put into building and maintaining a custom solution.

**Comment by tauhid621** [comment]
Recently, github released [required workflows](https://github.blog/2023-01-10-introducing-required-workflows-and-configuration-variables-to-github-actions/). Required workflows allows DevOps teams to define and enforce standard CI/CD practices across many source code repositories within an organization without needing to configure each repository individually

[Reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) can also be used to avoid duplication.

**Reply by gabrielbdornas** [reply]
> Recently, github released [required workflows](https://github.blog/2023-01-10-introducing-required-workflows-and-configuration-variables-to-github-actions/). Required workflows allows DevOps teams to define and enforce standard CI/CD practices across many source code repositories within an organization without needing to configure each repository individually
> 
> [Reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) can also be used to avoid duplication.

Hello everyone, any news about this feature?

The [Introducing required workflows and configuration variables to GitHub Actions](https://github.blog/enterprise-software/devops/introducing-required-workflows-and-configuration-variables-to-github-actions/) post was updated on June 12, 2023 saying it's not possible to use the created feature anymore to share actions across org's repo.

> Update: As of June 12, 2023, required workflows in GitHub Actions are now in limited beta enrollment and not available for new signups. See more information [here](https://docs.github.com/en/actions/using-workflows/required-workflows).

Is there any chance it could come back?

**Reply by maxforasteiro** [reply]
https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets it seems they moved to repo rulesets

**Comment by marina-chibizova** [comment]
ok, thank you everybody.
